scripts/convert.py

```diff
@@ -1,235 +1,241 @@
-
-        # last number regex. I know regex is hacky, but its reliablyhacky(tm).
-        self.imageidxre = re.compile(r'(\d+)(?!.*\d)')
-
-        for item in batch.iterator():
-            self.convert(converter, item)
-
-    def check_skipframe(self, filename):
-        try:
-            idx = int(self.imageidxre.findall(filename)[0])
-            return not any(map(lambda b: b[0]<=idx<=b[1], self.frame_ranges))
-        except:
-            return False
-
-    def check_skipface(self, filename, face_idx):
-        aligned_face_name = '{}_{}{}'.format(Path(filename).stem, face_idx, Path(filename).suffix)
-        aligned_face_file = Path(self.arguments.input_aligned_dir) / Path(aligned_face_name)
-        # TODO: Remove this temporary fix for backwards compatibility of filenames
-        bk_compat_aligned_face_name = '{}{}{}'.format(Path(filename).stem, face_idx, Path(filename).suffix)
-        bk_compat_aligned_face_file = Path(self.arguments.input_aligned_dir) / Path(bk_compat_aligned_face_name)
-        return aligned_face_file not in self.input_aligned_dir and bk_compat_aligned_face_file not in self.input_aligned_dir
-
-    def convert(self, converter, item):
-        try:
-            (filename, image, faces) = item
-
-            skip = self.check_skipframe(filename)
-            if self.arguments.discard_frames and skip:
-                return
-
-            if not skip: # process frame as normal
-                for idx, face in faces:
-                    if self.input_aligned_dir is not None and self.check_skipface(filename, idx):
-                        print ('face {} for frame {} was deleted, skipping'.format(idx, os.path.basename(filename)))
-                        continue
-                    image = converter.patch_image(image, face, 64 if "128" not in self.arguments.trainer else 128)
-                    # TODO: This switch between 64 and 128 is a hack for now. We should have a separate cli option for size
-
-            output_file = get_folder(self.output_dir) / Path(filename).name
-            cv2.imwrite(str(output_file), image)
-        except Exception as e:
-            print('Failed to convert image: {}. Reason: {}'.format(filename, e))
-
-    def prepare_images(self):
-        self.read_alignments()
-        is_have_alignments = self.have_alignments()
-        for filename in tqdm(self.read_directory()):
-            image = cv2.imread(filename)
-
-            if is_have_alignments:
-                if self.have_face(filename):
-                    faces = self.get_faces_alignments(filename, image)
-                else:
-                    print ('no alignment found for {}, skipping'.format(os.path.basename(filename)))
-                    continue
-            else:
-                faces = self.get_faces(image)
-            yield filename, image, faces
+import cv2

```

Comment:
There is new line problem in convert.py the diff shows all lines as changed while you just modified some. Can you plz check?

Reply:
@Clorr Yeah, really annoyingly I merged with upstream master just before @oatssss fixed his newline issue, so the issue in that code pulled into my repo. Now my Git says that I'm in sync with upstream, although I'm clearly not :/

This was late last night, so I went to bed. Going to try to fix this morning.

I would ask that people refrain from testing until I get another commit in. I will update

Reply:
No problem, this happens ;-)